### PR TITLE
Format Code

### PR DESCRIPTION
**Pull request by hsbasu**
- Using custom script to extract version from debian/changelog
- Fix tab spacing in metadata file

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project('mousam',
-          version: '1.0.2',
+          version: run_command('./parse_changelog',  '--show-field', 'Version', check: true).stdout().split('~')[0].strip('\n'),
```
Comment (amit9838) [inline meson.build:2]: I had a question here, is it taking the latest version from the release tag?

Comment (amit9838) [reply meson.build:2]: does parse changelog generates the logs automatically when flatpak is build?

**Review by amit9838 — COMMENTED** [review]
added comments

File: meson.build (hunk 1 above)
Comment (hsbasu) [reply meson.build:2]: > I had a question here, is it taking the latest version from the release tag?

Yes. It's reading the first line of the `debian/changelog` file to determine the version..

**Comment by hsbasu** [comment]
I built both snap and flatpak locally on my system. And tested the app. Both of them are working.

**Comment by hsbasu** [comment]
I think we should change the snap and flatpak workflow so that it always checks for any build errors on any branch and PR.

**Comment by amit9838** [comment]
I want to know that , lets say i'm building new version(1.0.3) and Im compiling the flatpak  or snap , how can i Change the version to 1.0.3,  or is it going to pick when it is released

**Comment by hsbasu** [comment]
It will pick the version automatically. You just need to update these files:
```
data/io.github.amit9838.mousam.appdata.xml.in
debian/changelog
```
After you have updated the `data/io.github.amit9838.mousam.appdata.xml.in`, I'll update the `debian/changelog` using the command `gbp dch --full -N <the new version in data/io.github.amit9838.mousam.appdata.xml.in> -D stable`

**Comment by hsbasu** [comment]
@amit9838 I would suggest using the `Squash and merge` option rather than using `Merge Commits`.

**Comment by amit9838** [comment]
> It will pick the version automatically. You just need to update these files:
> 
> ```
> data/io.github.amit9838.mousam.appdata.xml.in
> debian/changelog
> ```
> 
> After you have updated the `data/io.github.amit9838.mousam.appdata.xml.in` with a new version, I'll update the `debian/changelog` using the command `gbp dch --full -N <the new version in data/io.github.amit9838.mousam.appdata.xml.in> -D stable`.
> 
> Edit: I think you might also need to use `--since` option as well with `gbp dch`. Anyway. If you ping me after updating the `data/io.github.amit9838.mousam.appdata.xml.in`, I'll do the rest for you.

I tested this thing, but it is not building the app in runner in debug mode, and I think it not fesible to always build the project and generate new package. And i think there is one more issue of dependency of project over the changelog file in the deb dir(it need to run a script first) to make any new release.
So what i would suggest it to either create a new file may be VERSION.txt in deb dir containing version which can be updated manually whenever a new release is made,  or get it from meson file using some python script.
Example :
```python
import re

def get_package_version(file_path):
    with open(file_path, 'r') as file:
        content = file.read()
        version_match = re.search(r"version:\s*'([\d.]+)'", content)
        if version_match:
            return version_match.group(1)
        else:
            return None

meson_file_path = 'meson.build'
version = get_package_version(meson_file_path)
if version:
    print(f"The package version is: {version}")
else:
    print("Package version not found in the meson.build file.")

```

**Comment by hsbasu** [comment]
Which runner did you use and how?

**Comment by amit9838** [comment]
I mean Builder

**Comment by hsbasu** [comment]
> And i think there is one more issue of dependency of project over the changelog file in the deb dir(it need to run a script first) to make any new release.

Which script are you saying? Can you explain?

> So what i would suggest it to either create a new file may be VERSION.txt in deb dir containing version which can be updated manually whenever a new release is made, or get it from meson file using some python script.

Debian packaging does not work that way. For debian packaging, you create the `changelog` file first. Before you build a `.deb` package.

**Comment by hsbasu** [comment]
> I mean Builder https://flathub.org/apps/org.gnome.Builder

Give me some time. Leave the PR as it is. I'll test it with `gnome-builder`. Have you tried following flatpak packaging tutorials [here](https://docs.flatpak.org/en/latest/first-build.html)?

There they say that to build a flatpak you just need to run 
```
flatpak-builder --user --install --force-clean build-dir io.github.amit9838.mousam.json`
flatpak run io.github.amit9838.mousam
```
If these two steps does not show any issue, then everything's alright.

**Comment by hsbasu** [comment]
> I mean Builder https://flathub.org/apps/org.gnome.Builder

Can you paste the error that you are getting?

**Comment by amit9838** [comment]
Have you tried building using play button given in the header

**Comment by amit9838** [comment]
> > And i think there is one more issue of dependency of project over the changelog file in the deb dir(it need to run a script first) to make any new release.
> 
> Which script are you saying? Can you explain?
> 
> > So what i would suggest it to either create a new file may be VERSION.txt in deb dir containing version which can be updated manually whenever a new release is made, or get it from meson file using some python script.
> 
> Debian packaging does not work that way. For debian packaging, you create the `changelog` file first. Before you build a `.deb` package.

I mean this command, 
`gbp dch --full -N <the new version in data/io.github.amit9838.mousam.appdata.xml.in> -D stable`

So, you can create the changelog file, without modefing the meson.build file right?

**Comment by hsbasu** [comment]
> > > And i think there is one more issue of dependency of project over the changelog file in the deb dir(it need to run a script first) to make any new release.
> > 
> > 
> > Which script are you saying? Can you explain?
> > > So what i would suggest it to either create a new file may be VERSION.txt in deb dir containing version which can be updated manually whenever a new release is made, or get it from meson file using some python script.
> > 
> > 
> > Debian packaging does not work that way. For debian packaging, you create the `changelog` file first. Before you build a `.deb` package.
> 
> I mean this command, `gbp dch --full -N <the new version in data/io.github.amit9838.mousam.appdata.xml.in> -D stable`
> 
> So, you can create the changelog file, without modefing the meson.build file right?

Yes. Of course. I changed it to dynamic version so that the `meson.build` need not be modified every time we do a version update.

**Comment by hsbasu** [comment]
> Have you tried building using play button given in the menubar (center)

Yes. I clicked on that to build. Actually, gnome-builder is just an IDE like VS Code. It's not necessary to build flatpak or snaps. And IDE, you can use anything, be it gnome-builder or Code or Atom.

**Comment by amit9838** [comment]
Ok, so deb build entirly depends on changelog file. So lets remove dynamic versioning from. For now we can change it manually on new release. W'll fix it later.

**Comment by hsbasu** [comment]
I'll remove the versioning part from this PR.

**Comment by hsbasu** [comment]
@amit9838 I updated this commit and now it only fixes the tab spacing in the metadata file.

**Review by amit9838 — APPROVED** [review]
Thnaks, LGTM

**Pull request merged**